doc/source/serve/advanced-guides/multi-app-container.md

```diff
@@ -0,0 +1,175 @@
+Similarly, `resnet.Dockerfile` uses the appropriate base image, installs necessary dependencies, and downloads the resnet model source code.
+Save the following to files named `whisper.Dockerfile` and `resnet.Dockerfile`.
+
+::::{tab-set}
+:::{tab-item} whisper.Dockerfile
+```dockerfile
+FROM rayproject/ray:latest-py38-gpu
+
+RUN pip install faster_whisper==0.10.0
+RUN sudo apt-get update && sudo apt-get install curl -y
+
+RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/whisper_example.py
+```
+:::
+:::{tab-item} resnet.Dockerfile
+```dockerfile
+FROM rayproject/ray:latest-py38-cpu
+
+RUN torch==2.0.1 torchvision==0.15.2
+RUN sudo apt-get update && sudo apt-get install curl -y
+
+RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/resnet50_example.py
+```
+:::
+::::
+
+Then, build the corresponding docker images and push it to your choice of docker registry.
+
+::::{tab-set}
+:::{tab-item} Whisper
+```bash
+export IMG1={your-image-repo-and-name}
+docker build -t $IMG1 -f whisper.Dockerfile .
+docker push $IMG1
+```
+:::
+:::{tab-item} Resnet
+```bash
+export IMG2={your-image-repo-and-name}
+docker build -t $IMG2 -f resnet.Dockerfile .
+docker push $IMG2
+```
+:::
+::::
+
+Finally, you can specify the docker image within which you want to run each application in the `container` field of an application's runtime environment specification. The `container` field has three fields:
+- `image`: (Required) The image to run your application in.
+- `worker_path`: The absolute path to `default_worker.py` inside the container. This is necessary if the Ray installation directory is different than that of the host (where raylet is running). For instance, if you are running on bare metal, and using a standard Ray docker image as the base image for your application's, then the Ray installation directory on host will likely differ from that of the container. Note that if you are using a standard Ray docker image as the base image, then `default_worker.py` will always be found at `/home/ray/anaconda3/lib/{python-version}/site-packages/ray/_private/workers/default_worker.py`.
+- `run_options`: Additional options to pass to the `podman run` command used to start a Serve deployment replica in a container. See [podman run documentation](https://docs.podman.io/en/latest/markdown/podman-run.1.html) for a list of all options.
+
+The following Serve config will run the `whisper` app with the image `your-image-1`, and the `resnet` app with the image `your-image-2`. Concretely, this means that all deployment replicas in the applications will start and run in containers with the respective images.
+
+```yaml
+applications:
+  - name: whisper
+    import_path: whisper_example:entrypoint
+    route_prefix: /whisper
+    runtime_env:
+      container:
+        image: {your-image-1}
+        worker_path: /home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py
```

Comment:
hopefully this is optional. it's very ugly, exposing a private API (the organization of internals of our python package)

Reply:
why is this strictly necessary? Can't we do an `import ray` and find the path dynamically?

Reply:
at a minimum, we should change this to only be the `/home/ray/anaconda3/lib/python3.8/site-packages/ray/` prefix so it isn't depending on the structure of the files in the `ray` package

Reply:
Agreed, this is pretty ugly. This is leftover from the initial implementation of this plugin, and I didn't have a chance to refine it. I'll focus on improving this UX in the upcoming release.